core/src/smartcontracts/isi/domain.rs

```diff
@@ -61,8 +58,9 @@ pub mod isi {
             wsv: &WorldStateView<W>,
         ) -> Result<(), Self::Error> {
             let account_id = self.object_id;
+
             wsv.modify_domain(&account_id.domain_id, |domain| {
-                domain.accounts.remove(&account_id);
+                assert!(domain.remove_account(&account_id).is_some());
```

Comment:
Is it okay to `assert!` without a prior check like in `Register`?

Reply:
I'll add checks when removing

Reply:
the question here is - should remove operations be no-op when the entity isn't present or should an error be returned? To my mind returning an error at least gives some feedback to the user about what happened

Reply:
I've just noticed, in the validation [the execution and the permission check](https://github.com/mversic/iroha/blob/7d1dfebb1b7097a7ec661d4f03a00666c40ffe23/core/src/tx.rs#L117-L135) should be reversed.
Otherwise, we are letting someone forbidden know about the (non)existence of the object

Reply:
oh yeah. and also someone could issue an instruction to change permission and the permission check would pass

Reply:
We have very few idempotent instructions. The operation should return an error.

Reply:
yeah, will log or return errors